ICT4DNews/app/src/main/java/at/ict4d/ict4dnews/screens/news/list/ICT4DNewsFragment.kt

```diff
@@ -59,41 +60,77 @@ class ICT4DNewsFragment : BaseNavigationFragment<ICT4DNewsViewModel, FragmentIct
             model.requestToLoadJsonFeed()
         }
 
+        model.searchedNewsList.observe(this, Observer {
+            if (it != null) {
+                Timber.d("Search result size is ----> ${it.size}")
+                adapter.submitList(it)
+            }
+        })
+
         binding.recyclerview.layoutManager = LinearLayoutManager(context)
         binding.recyclerview.adapter = adapter
 
         model.newsList.observe(this, Observer {
             if (it != null && it.isNotEmpty()) {
-                Timber.d("list in fragment: ${it.size}")
-                adapter.submitList(it)
+                if (model.searchQuery == null) {
+                    Timber.d("list in fragment: ${it.size}")
+                    adapter.submitList(it)
+                }
             }
         })
 
         return view
     }
 
     override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
+        this.menu = menu
         inflater?.inflate(R.menu.ict4dnews_menu, menu)
         val menuItem = menu?.findItem(R.id.menu_search)
         val searchView = menuItem?.actionView as SearchView
 
         compositeDisposable.add(RxSearchView.queryTextChanges(searchView)
             .debounce(400, TimeUnit.MILLISECONDS)
             .map { char: CharSequence -> char.toString() }
-            .filter { query: String -> !query.isEmpty() }
             .observeOn(Schedulers.io())
             .subscribeOn(AndroidSchedulers.mainThread())
             .subscribe({ query ->
                 Timber.d("*** query: $query")
-                activity?.runOnUiThread {
-                    activity?.toast("You will search for: $query")
-                }
+                model.performSearch(query)
             }, { e ->
                 Timber.e("*** error: $e")
             }, {
                 Timber.d("*** complete")
             })
         )
+
+        // restore search view after orientation change
+        model.searchQuery?.let {
+            if (it.isNotEmpty()) {
+                enableRefreshMenuItem(false)
+                menuItem.expandActionView()
+                searchView.setQuery(it, true)
+                searchView.clearFocus()
+            }
+        }
+
+        menuItem.setOnActionExpandListener(object : MenuItem.OnActionExpandListener {
+            override fun onMenuItemActionExpand(p0: MenuItem?): Boolean {
+                enableRefreshMenuItem(false)
+                return true
+            }
+
+            override fun onMenuItemActionCollapse(p0: MenuItem?): Boolean {
+                adapter.submitList(model.searchedNewsList.value)
+                model.searchQuery = null
+                enableRefreshMenuItem(true)
+                return true
+            }
+        })
+    }
+
+    private fun enableRefreshMenuItem(isEnable: Boolean) {
```

Comment:
change to variable name to `isEnabled` or `enable` - otherwise it's not English 😉